### PR TITLE
WIP: Add on-demand integration test workflow

### PR DESCRIPTION
**Pull request by oxkitsune**

### CODE REVIEW

**Comment by oxkitsune** [comment]
testing the /integration-test command!

**Comment by oxkitsune** [comment]
testing the /integration-test command

**Comment by oxkitsune** [comment]
testing the /integration-test command

**Comment by oxkitsune** [comment]
testing the /integration-test command

**Comment by oxkitsune** [comment]
testing the /integration-test command

**Comment by oxkitsune** [comment]
testing the /integration-test command

**Comment by oxkitsune** [comment]
testing the /integration-test command

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    #   github.event.issue.pull_request &&
+    #   contains(github.event.comment.body, '/integration-test')
+    # XXX TODO: Device matrix? 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1
+        with:
+          submodules: 'recursive'
+          # ensure we fetch ALL commits and tags!
```
Comment (rickie) [inline .github/workflows/integration-test.yml:28]:
```suggestion
          # Ensure that all commits and tags are being fetched.
```

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+jobs:
+  on_demand_integration_test:
+    name: On-demand integration test
+    # Only run the integration test if the created comment is on a pull request
+    # and contains the string `/integration-test`
+    # disable this to avoid having to change the default branch on EPS
+    # if: | 
+    #   github.event.issue.pull_request &&
+    #   contains(github.event.comment.body, '/integration-test')
+    # XXX TODO: Device matrix?
```
Comment (rickie) [inline .github/workflows/integration-test.yml:21]: Also not for now I'd say, let's omit.

File: .github/workflows/integration-test.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: On-demand integration test
+on:
+  # run action every time a new comment is created
+  issue_comment:
+    types: [ created ]
+  # enable this, to test the action without changing default branch
```
Comment (rickie) [inline .github/workflows/integration-test.yml:6]:
```suggestion
  # XXX: Enable this, to test the action without changing default branch.
```

**Review by rickie — COMMENTED** [review]
Some feedback, I know you are going to push some more changes still.

**Comment by Stephan202** [comment]
This PR is superseded by #865. Primary improvements in that PR:
1. A more robust integration test script that runs to completion without error.
2. Build artifacts are uploaded only on test failure (rather than only on success).

**Pull request closed without merging**